### PR TITLE
libteec: Convert ioctl errno to an appropriate TEEC error

### PR DESCRIPTION
**Pull request by Sadiq1811**
Signed-off-by: Sadiq Hussain <sadiq.muchumarri@intel.com>

### CODE REVIEW

File: libteec/src/tee_client_api.c [hunk 1]
```diff
@@ -514,12 +514,43 @@ static void teec_free_temp_refs(TEEC_Operation *operation,
 
 static TEEC_Result ioctl_errno_to_res(int err)
 {
-	switch (err) {
-	case ENOMEM:
-		return TEEC_ERROR_OUT_OF_MEMORY;
-	default:
-		return TEEC_ERROR_GENERIC;
-	}
+    switch (err) {
```
Comment (jenswi-linaro) [inline libteec/src/tee_client_api.c:517]: I think you're taking this too far. These are the codes that the driver normally uses:
E2BIG, EBUSY, ECHILD, EFAULT, EINVAL, ENODEV, ENOENT, ENOMEM, ENOTSUPP, ENXIO, EOPNOTSUPP, EPERM, ERESTARTSYS
Most of those probably make sense to just report as `TEEC_ERROR_GENERIC`. I'd rather focus on the few which we care about, I guess that's `ENOMEM` and `EINVAL` for now.

Comment (Sadiq1811) [reply libteec/src/tee_client_api.c]: Thanks for the review comment.
I have pushed the updated change now.

**Comment by jenswi-linaro** [comment]
`Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>`

**Comment by Sadiq1811** [comment]
@jenswi-linaro anything else to be done from side?

**Comment by jenswi-linaro** [comment]
Please add the tag to your commit message and force push again.

**Comment by Sadiq1811** [comment]
So, the tag would be named "3.15.0"?

**Comment by jenswi-linaro** [comment]
No, just add the line:
```
Reviewed-by: Jens Wiklander <jens.wiklander@linaro.org>
```
to the commit message.

**Comment by Sadiq1811** [comment]
Done

**Pull request merged**